### ISSUE TITLE
Добавление обработчика события в пустую область удаляет ее. А должно наоборот создавать необходимую область. Помещаться в правильную область.

### ISSUE THREAD

**Issue by DmitryShvaika** [1C-Company/1c-edt-issues#1525]
### Описание ошибки

Добавление обработчика события в пустую область удаляет ее. А должно наоборот создавать необходимую область. Помещаться в правильную область.

![Добавление события удаляет область](https://github.com/user-attachments/assets/a9e22b2f-79ab-4d8e-bc97-e56f12b6c1f1)

### Как воспроизвести

1. Повторить действия с видео

### Ожидаемое поведение

Область не должна удалятся.
Если области нет, должна создаваться.
Если область есть и пытаются добавить не в ту область, добавлять в правильную.

### Лог рабочей области

Не требуется.

### Версия 1С:EDT

Ruby 2024.1.0

### Операционная система

Windows

### Версия платформы 1С:Предприятие 8

8.3.21.1644

**Comment by MaksimDzyuba** [comment]
Здравствуйте, а скажите, пожалуйста, а у Вас не выключена ли опция по созданию структуры модуля из плагина по разработкам по стандартам
![image](https://github.com/user-attachments/assets/f158bf0f-dc88-4690-84c7-dfbb79b1a837)

**Comment by DmitryShvaika** [comment]
Нет, опция выключена. Она вроде бы отключена даже по умолчанию. Если дело в этом, то можно закрыть задачу. Попробую использовать эту настройку.

**Comment by MaksimDzyuba** [comment]
Ага, я понял, мы еще обсудим внутри, сообщим о решении, не очень, наверное, хорошо, что пропадает кусок кода, который в принципе остаться должен был, но обработчик да, создастся не в нем.

**Comment by DmitryShvaika** [comment]
При включенной галке автоматически создавать структуру модуля.
Дело в том, что код не сразу весь пишется. Обработчики иногда удаляются, иногда добавляются...
Вот, например создаю форму со включенной галкой создать структуру модуля. Но мне нужен только обработчик команды.
![image](https://github.com/user-attachments/assets/76ab3240-69a4-43d8-aee0-1158a11a0667)

Удалю лишние области. Смысл их вообще добавлять сразу?
Создаю команду. Она добавляется в правильную область.

![image](https://github.com/user-attachments/assets/d0ecb361-53ec-4743-98b9-0a6d17e9a518)

Пустые области я не могу оставить. Поэтому они все удалены.
Далее мне, например, на другой день, понадобилось создать обработчик ПриСозданииНаСервере.
![Создание областей](https://github.com/user-attachments/assets/7f96935c-348a-453d-964b-bd47bb31b75f)

Область создается, но опять как-то не очень.
Мне кажется, поведение по галке автоматически создавать структуру модуля должно быть немного иным (возможно и настройку лучше переименовать).
Не нужно сразу генерить кучу пустых областей, т.к. потом их необходимо удалять. Лучше создавать модуль пустым, но при добавлении обработчика помещать его в нудную область.

Еще, хотелось бы задавать форматирование областей: отступ между областями, отступ от границы внутри области.

**Comment by Kudesunik** [comment]
Добрый день!

> Понадобилось создать обработчик ПриСозданииНаСервере... Область создается, но опять как-то не очень.

Это баг, будет принято в работу на исправление. Пока что, чтобы избежать подобного поведения, можно удалить все пустые строки после //strict-types и перед любой областью или процедурой.

> Рефакторинг - извлечение метода так же не хочет создавать область и правильно помещать в него метод. В данном случае процедура Тест должна была попасть в область СлужебныеПроцедурыИФункции.

Это недоработка, будет принято в работу на исправление.
